### PR TITLE
Don't validate jdk when validating the cloud sdk

### LINKED ISSUE #441: Should CloudSDK validation be coupled with jdk validation?
[CloudSdk#validateCloudSdk](https://github.com/GoogleCloudPlatform/appengine-plugins-core/blob/713bff4c658b62640345e6312f59089c0e2c3437/src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java#L469) also validates that the system jdk points to a proper executable. 

Should these be coupled like this? This causes all validation of the CloudSDK to fail on improper jdk config. Presumably not all usages of the SDK require this.

### PR DESCRIPTION
**Pull request by etanshaul**
Fixes #441 

Removes jdk validation from the SDK validation method.

I did _not_ remove the jdk validation entirely:

It is still invoked when running an appCfg command or a devAppServer1 command

### CODE REVIEW

**Comment by etanshaul** [comment]
@elharo could you take a look as well to see if/how this would affect eclipse?

**Comment by elharo** [comment]
Is this still useful? Can we get it in?

**Comment by loosebazooka** [comment]
@etanshaul  We still want this, yeah?

**Comment by etanshaul** [comment]
I think this has implications for the clients. Let me check and refresh my memory on this.

**Comment by etanshaul** [comment]
@loosebazooka this change still makes sense, and is ok from the IntelliJ side.

If it still is ok from for Eclipse then we can merge it in. @elharo @chanseokoh

**Comment by loosebazooka** [comment]
@TadCordle would we also need to update our sdk checker in build tools?

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java [hunk 1]
```diff
@@ -506,7 +506,6 @@ public void validateCloudSdk()
           InvalidJavaSdkException {
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: Need to remove `InvalidJavaSdkException`.

**Comment by TadCordle** [comment]
> @TadCordle would we also need to update our sdk checker in build tools?

If we still want jdk validation, then yes. If we still want to do validation, is the only way of doing that now running an appCfg command or a devAppServer1 command and seeing if it throws an exception?

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java (hunk 1 above)
Comment (etanshaul) [reply src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: done

**Comment by etanshaul** [comment]
when would you want to call `validateJdk`? @loosebazooka

**Comment by etanshaul** [comment]
@chanseokoh sorry, pushed a minor cleanup which dismissed your approval

**Comment by etanshaul** [comment]
@ehlaro, yes, please approve and I'll merge it in after CI is complete. thanks

**Comment by etanshaul** [comment]
@elharo actually looks like its already approved. never mind. will merge.

**Pull request merged**